### DISCUSSION TITLE
Using 'select' with a signal data structure for live-updates

### DISCUSSION THREAD

**Discussion by qknight** [leptos-rs/leptos#1919] [General]
I have this problem:

https://stackoverflow.com/questions/77327851/leptos-update-the-option-text-when-the-underlaying-datastructure-changes

Please direct me to a example where the <option> inside a <select>...</select> are created and updated dynamically based on the _selectionOptions_ signal below:

    let selectionOptions1: Vec<SelectionOption> = vec![
        SelectionOption{ label: "h0rses", id: 0, amount: 0},
        SelectionOption{ label: "b1rds", id: 1, amount: 10},
        SelectionOption{ label: "2nfish", id: 2, amount: 0},
        SelectionOption{ label: "3lk", id: 3, amount: 20},
    ];
    let selectionOptions = create_rw_signal::<Vec<SelectionOption>>(selectionOptions1);

I'm completely lost. Seems the For does not update the structure on demand.

**Comment by gbj** [comment]
I'm not sure what this attribute on the `<option>` is supposed to do:
```
<option /* */
  default_selection = (default_selection.get() == Some(option.id))
```
Do you mean to use the `selected` attribute?
```
<option /* */
  selected=(move || default_selection.get() == Some(option.id))
```
(I added the closure so it would be reactive)

**Comment by qknight** [comment]
I rewrote the code as this:

From

    pub amount: u32,

To

    pub amount: RwSignal<u32>,

And the For loop from:

    <For
      each = move || selectionOptions.get()
      key = |option| option.id
      let:option
    >
    <option
      value = option.id
      default_selection = (default_selection.get() == Some(option.id))
      > {
      let v = format!("{} - {}", option.label, option.amount);
      v
      }
    </option>
    </For>

To

    <For
      each = {move || selectionOptions.clone().get()}
      key = |option| option.id
      let:option
    >
    <option
      value = move || option.id
      default_selection = (default_selection.get() == Some(option.id))
      > { move || {
      let z = option.amount.get();
      let v = format!("{} - {}", option.label, z);
      v
      }
      }
     </option>
   </For>

So now the `select` UI element's text, i.e. footext in

    <option value="0">footext</option>

Is actually updated once the underlaying vector is modified. I still got problems with the code section as there is some borrowing wrong but maybe I'll figure that out as well.

Thanks for your help!
